### PR TITLE
Feature 90545: SSO Authentication

### PR DESCRIPTION
**Pull request by desousamario85**
Added the SSO function and Homescreen for users to login.

### CODE REVIEW

**Comment by BenGWeeks** [comment]
Can you give a little more context, perhaps some screenshots? :-)

Seems to be some change to the LNBits Service and `FeedList` as well included in this?

**Comment by desousamario85** [comment]
Changes in the Feedlist was to fix the issue I had with the looping of the wallets, which caused 429 error.

**Comment by BenGWeeks** [comment]
> Changes in the Feedlist was to fix the issue I had with the looping of the wallets, which caused 429 error.

OK, can you check this now works for you. Both `main` and `react-web-app` are working for me and @EdiWeeks.

**Comment by desousamario85** [comment]
When I resynced the branch the feed was working.

File: tabs/src/ui-components/PageLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+
+
+type Props = {
+    children?: React.ReactNode;
+};
+
+export const PageLayout: React.FC<Props> = ({children}) => {
+    return (
+        <>
```
Comment (BenGWeeks) [inline tabs/src/ui-components/PageLayout.tsx:9]: What is this `<>`?

File: tabs/src/components/Home.tsx
Comment (BenGWeeks) [inline tabs/src/components/Home.tsx:1]: What is this page?

**Review by BenGWeeks — CHANGES_REQUESTED** [review]
See inline comments.

Also, the demo didn't go to the regular Home (with Feed) screen.

File: feed.png
Comment (BenGWeeks) [inline feed.png:1]: Let's remove this.

File: tabs/.gitignore [hunk 2]
```diff
@@ -17,6 +17,7 @@
 .env.development.local
 .env.test.local
 .env.production.local
+.env.development
```
Comment (BenGWeeks) [inline tabs/.gitignore:20]: Happy to include this.

File: tabs/src/components/graph.tsx
[the remark this answers is not in the document]
Comment (BenGWeeks) [reply tabs/src/components/graph.tsx:1]: Rename to "UserDetailsComponent"

File: .github/workflows/master_zappie-dev.yml [hunk 3]
```diff
@@ -3,12 +3,12 @@ name: Build and deploy Node.js app to Azure Web App - zappie-dev
 on:
   push:
     branches:
-      - master
+      - SSO
```
[the remark this answers is not in the document]
Comment (desousamario85) [reply .github/workflows/master_zappie-dev.yml:6]: The Workflow won't run for deployment as there is no SSO repo here. But this doesn't affect the application

File: tabs/src/App.tsx [hunk 4]
```diff
@@ -1,22 +1,51 @@
...
+                </PageLayout>
+            </ThemeProvider>
+        </MsalProvider>
+    );
+}
+
+function Pages() {
+    return (
+        <Routes>
+            <Route path="/test" element={<FeedComponent />} />
```
[the remark this answers is not in the document]
Comment (desousamario85) [reply tabs/src/App.tsx:45]: Yes, the homepage file is in the component directory.  We will need to create a pages directory to store all our pages to be separate from the components.

File: tabs/src/Home.tsx [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import FeedComponent from './components/FeedComponent';
+import FeedComponent from './components/test';
```
[the remark this answers is not in the document]
Comment (desousamario85) [reply tabs/src/Home.tsx:2]: updated the back to the original feed component.

File: tabs/package.json [hunk 6]
```diff
@@ -3,7 +3,14 @@
...
   "dependencies": {
-    "@fluentui/react": "^8.120.6",
+    "@azure/msal-browser": "^3.23.0",
+    "@azure/msal-react": "^2.0.22",
+    "@emotion/react": "^11.13.3",
+    "@emotion/styled": "^11.13.0",
+    "@fluentui/react": "^8.120.9",
+    "@fluentui/react-components": "^9.54.14",
+    "@mui/icons-material": "^6.0.2",
+    "@mui/material": "^6.0.2",
```
[the remark this answers is not in the document]
Comment (desousamario85) [reply tabs/package.json:13]: still have some components MUI which will be converted in the upcoming PRs

**Comment by desousamario85** [comment]
Made some admendments to the PR code.

File: tabs/src/App.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (BenGWeeks) [reply tabs/src/App.tsx:45]: The pages should be in the src folder (as they are currently), with components in the components folder.

**Comment by BenGWeeks** [comment]
> 
image

This doesn't look as faded as in Figma?

File: tabs/package.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (BenGWeeks) [reply tabs/package.json:13]: It's easier to remove it if we never put it in.

File: tabs/src/components/FeedList.tsx
Comment (BenGWeeks) [inline tabs/src/components/FeedList.tsx:1]: I think this needs to be removed from the PR.

File: tabs/src/components/Home.tsx
Comment (BenGWeeks) [inline tabs/src/components/Home.tsx:1]: Have have we moved Home.tsx to the Components folder?

File: tabs/src/components/graph.tsx
Comment (BenGWeeks) [inline tabs/src/components/graph.tsx:1]: Rename to `GraphComponent` and ensure a small simple component that can be added on to other pages if needed for testing / demo.

File: tabs/src/components/test.tsx
Comment (BenGWeeks) [inline tabs/src/components/test.tsx:1]: Remove this.

File: tabs/src/services/lnbitsServiceLocal.ts [hunk 7]
```diff
@@ -453,8 +453,8 @@ async function ensureMatchingUserWallet(
   try {
     //const apiKey = await getAccessToken(userName, password); // Assuming getAccessToken returns the API key
 
-    let walletName = null;
-    if (walletType == 'Sending') {
+    let walletName =  null;
```
Comment (BenGWeeks) [inline tabs/src/services/lnbitsServiceLocal.ts:456]: Introduced unneeded space.

File: tabs/src/services/lnbitsServiceLocal.ts [hunk 8]
```diff
@@ -500,6 +500,38 @@ const getWalletIdByUserId = async (adminKey: string, userId: string) => {
   }
 };
 
+const getTotalSentAmount = async (inKey: string): Promise<number> => {
```
Comment (BenGWeeks) [inline tabs/src/services/lnbitsServiceLocal.ts:503]: getPaymentsTotal would be a better title?

Should return a Promise.

File: tabs/src/ui-components/ErrorComponent.tsx
Comment (BenGWeeks) [inline tabs/src/ui-components/ErrorComponent.tsx:1]: Why have you created a new `ui-components` folder? We already have a components folder.

File: tabs/src/ui-components/Loading.tsx
Comment (BenGWeeks) [inline tabs/src/ui-components/Loading.tsx:1]: Move to `/Components`

File: tabs/src/ui-components/PageLayout.tsx
Comment (BenGWeeks) [inline tabs/src/ui-components/PageLayout.tsx:1]: Not sure what this is doing.

If needed, move to `/Components`

File: tabs/src/ui-components/SignInButton.tsx
Comment (BenGWeeks) [inline tabs/src/ui-components/SignInButton.tsx:1]: Move to `/Components`

Comment (BenGWeeks) [reply tabs/src/ui-components/SignInButton.tsx:1]: Do we need this button if we have a `SignInSignOutButton` too?

File: tabs/src/ui-components/SignOutButton.tsx
Comment (BenGWeeks) [inline tabs/src/ui-components/SignOutButton.tsx:1]: Do we need this if we have a `SignInSignOutButton`?

File: tabs/src/utils/MsGraphApiCall.tsx
Comment (BenGWeeks) [inline tabs/src/utils/MsGraphApiCall.tsx:1]: Is this some test code?

File: tabs/src/ui-components/WelcomeName.tsx
Comment (BenGWeeks) [inline tabs/src/ui-components/WelcomeName.tsx:1]: Is this test code?

File: tabs/src/utils/NavigationClient.tsx
Comment (BenGWeeks) [inline tabs/src/utils/NavigationClient.tsx:1]: We already have `FooterComponent` that we discussed for this?

Comment (desousamario85) [reply tabs/src/utils/NavigationClient.tsx:1]: This is part of the SSO. not part of the navigation bar or footer.

File: tabs/src/utils/MsGraphApiCall.tsx
Comment (desousamario85) [reply tabs/src/utils/MsGraphApiCall.tsx:1]: no , this is not test code. It is displaying the logged in users name which was part of the top navigation. Footers don't normally display logged in users details.

File: tabs/src/ui-components/SignOutButton.tsx
Comment (desousamario85) [reply tabs/src/ui-components/SignOutButton.tsx:1]: Yes, as it switches between them. part of the SSO steps of congifuration

File: tabs/src/ui-components/SignInButton.tsx
Comment (desousamario85) [reply tabs/src/ui-components/SignInButton.tsx:1]: the files is required.

File: tabs/src/ui-components/PageLayout.tsx (hunk 1 above)
Comment (desousamario85) [reply tabs/src/ui-components/PageLayout.tsx:9]: It shows all the children items. In the components pages. Meaning this is the master pageout. and everything else falls inbetween it. So we add the Nav/Footer component here and then it displays on all pages, without having to add the nav/footer component on every page.

File: tabs/src/ui-components/ErrorComponent.tsx
Comment (desousamario85) [reply tabs/src/ui-components/ErrorComponent.tsx:1]: part of the process of the SSO steps. I'll move everything to Components

**Comment by BenGWeeks** [comment]
Pushed some updates. Raised a couple of bugs, main one being devs can't use it from localhost:3000.

**Comment by desousamario85** [comment]
I have added the steps for the SSO to configure the bot client ID in the Azure Portal. This will allow you to test locally using localhost. The ReadMe was updated in the Tabs folder.

File: tabs/README.md [hunk 9]
```diff
@@ -16,6 +16,19 @@ NB We install version `2.0.6` because that is the version used by `webpack-dev-s
 npm ls http-proxy-middleware
 ```
 
+
+## SSO Authentication
+
+For the SSO authentication function to work, obtain the MS Bot Framework ID saved in the Environment variables.
+1. Navigate to https:portal.azure.com and search for App Registrartion
```
Comment (BenGWeeks) [inline tabs/README.md:23]: Typo

**Comment by BenGWeeks** [comment]
Needs merge conflicts resolved @desousamario85 please :-)

**Comment by desousamario85** [comment]
All conflicts resolved.

**Comment by BenGWeeks** [comment]
- `components/ZapsChart` should be removed (already implemented and not concerning this PR)
- `ZapsSents` should be moved to a different PR
- `interfaces/DataModel,.ts` was removed a while ago (we're using `global.d.ts`).
- `mui` is still included.
- I don't see any reason why `global.d.ts` should be changed as part of this PR.
- Your description associated with this PR should give more information about the PR other than "Added the SSO function and Homescreen for users to login." - needs to summarize the PR, what I need to do to test it, etc.
- Your 56 commits should be squashed down to a few logical ones.

**Comment by BenGWeeks** [comment]
When it's ready for testing, please let me know. For now, setting this back to Draft as I keep checking it after a commit, thinking it's ready.

**Pull request closed without merging**